frontend/lib/activation/deeplink_activation.dart

```diff
@@ -103,41 +103,41 @@ class _DeepLinkActivationState extends State<DeepLinkActivation> {
                 child: Column(mainAxisSize: MainAxisSize.min, children: [
                   if (_state == _State.waiting) _WarningText(status, userCodeModel),
                   ElevatedButton.icon(
-                    onPressed: activationCode != null &&
-                            _state == _State.waiting &&
-                            status == DeepLinkActivationStatus.valid
-                        ? () async {
-                            setState(() {
-                              _state = _State.loading;
-                            });
-                            try {
-                              final activated = await activateCard(context, activationCode);
-                              if (!context.mounted) return;
-                              if (activated) {
-                                final cardAmount = Provider.of<UserCodeModel>(context, listen: false).userCodes.length;
-                                GoRouter.of(context).pushReplacement('$homeRouteName/$identityTabIndex/$cardAmount');
+                    onPressed:
+                        activationCode != null && _state == _State.waiting && status == DeepLinkActivationStatus.valid
+                            ? () async {
                                 setState(() {
-                                  _state = _State.success;
-                                });
-                              } else {
-                                setState(() {
-                                  _state = _State.waiting;
+                                  _state = _State.loading;
                                 });
+                                try {
+                                  final activated = await activateCard(context, activationCode);
+                                  if (!context.mounted) return;
+                                  if (activated) {
+                                    final cardIndex =
```

Comment:
Looks like some ugly formatting issues happened.
Only changed LN 116 and LN117
to use the correct card index